components/org.wso2.carbon.identity.tag.manager.core/src/main/java/org/wso2/carbon/identity/tag/manager/core/dao/TagManagementDAOImpl.java

```diff
@@ -0,0 +1,361 @@
+            prepStmt.setString(3, tenantUuid);
+            resultSet = prepStmt.executeQuery();
+            if (resultSet.next()) {
+                tag = new Tag();
+                tag.setTagUuid(resultSet.getString(1));
+                tag.setDescription(resultSet.getString(2));
+                tag.setPubliclyVisible(isPublic(resultSet.getString(3)));
+            }
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_LOADING_TAG.getCode(), e);
+        }
+        return tag;
+    }
+
+    @Override
+    public TagListResult filterTags(int limit, int offset, String sortBy, String sortOrder, String filter,
+                                    String tenantUuid) throws TagServiceException {
+
+        if (limit == 0) {
+            return null;
+        }
+        PreparedStatement prepStmt;
+        ResultSet resultSet;
+        TagListResult tagListResult = new TagListResult();
+        List<Tag> tagList = new ArrayList<>();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+
+            String filterResolvedForSQL = resolveSQLFilter(filter);
+
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.LOAD_IDN_TAGS_BY_TYPE_NAME_TENANT_UUID);
+            prepStmt.setString(1, tenantUuid);
+            prepStmt.setString(2, filterResolvedForSQL);
+            prepStmt.setInt(3, limit);
+            prepStmt.setInt(4, offset);
+            resultSet = prepStmt.executeQuery();
+            while (resultSet.next()) {
+                Tag tag = new Tag();
+                tag.setTagUuid(resultSet.getString(1));
+                tag.setName(resultSet.getString(2));
+                tag.setDescription(resultSet.getString(3));
+                tag.setPubliclyVisible(isPublic(resultSet.getString(4)));
+                tagList.add(tag);
+            }
+            tagListResult.setTagList(tagList);
+            tagListResult.settotalAvailableResultss(tagList.size());
+        } catch (SQLException e) {
+            throw new TagServiceException(ErrorMessage.ERROR_CODE_ERROR_GETTING_TAGS_FROM_TYPE_TENANT_UUID.getCode(),
+                    e);
+        }
+        return tagListResult;
+    }
+
+    @Override
+    public List<String> loadTagTypes(int limit, int offset, String sortBy, String sortOrder)
+            throws TagServiceException {
+
+        PreparedStatement prepStmt;
+        ResultSet resultSet;
+        List<String> tagTypesList=null;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            prepStmt = connection.prepareStatement(Constants.SQLConstants.LOAD_IDN_TAG_TYPES);
+            prepStmt.setInt(1, limit);
+            prepStmt.setInt(2, offset);
+            resultSet = prepStmt.executeQuery();
+            if (resultSet.next()) {
+                tagTypesList.add(resultSet.getString(1));
+            }
+        } catch (SQLException | NullPointerException e) {
```

Comment:
NullPointerException is a run-time exception which is not recommended to catch it, but instead we need to avoid it

Reply:
I'll fixt it

Reply:
fixed